src/test/java/org/assertj/core/api/BaseAssertionsTest.java

```diff
@@ -67,15 +69,16 @@ public int compare(Method method1, Method method2) {
         // static vs not static is not important Soft vs Not Soft assertions
         boolean equal = (ACCESS_MODIFIERS & method1.getModifiers() & method2.getModifiers()) != 0;
         equal = equal && (ignoreDeclaringClass || method1.getDeclaringClass().equals(method2.getDeclaringClass()));
-        equal = equal && (ignoreReturnType || method1.getReturnType().equals(method2.getReturnType()));
+        equal = equal && (ignoreReturnType || canonize(method1.getGenericReturnType())
+          .equals(canonize(method2.getGenericReturnType())));
         equal = equal && (ignoreMethodName || method1.getName().equals(method2.getName()));
```

Comment:
Completely agree with you. I'll do that. I will also clean it up a bit. For example `ignoreDeclaringClass` is always `true`.